tests/phpunit/tests/Admin/Cross_Origin_Isolation.php

```diff
@@ -0,0 +1,228 @@
+		$html      = "<img src='/test.jpg' alt='test' />";
+		$attribute = 'src';
+		$url       = '/test.jpg';
+		$result    = $this->call_private_method( $object, 'add_attribute', [ $html, $attribute, $url ] );
+		$this->assertNotContains( 'crossorigin', $result );
+	}
+
+	/**
+	 * @covers ::starts_with
+	 */
+	public function test_starts_with() {
+		$object       = $this->get_coi_object();
+		$string       = 'hello world';
+		$start_string = 'hello';
+		$result       = $this->call_private_method( $object, 'starts_with', [ $string, $start_string ] );
+		$this->assertTrue( $result );
+	}
+
+	/**
+	 * @covers ::starts_with
+	 */
+	public function test_starts_with_fail() {
+		$object       = $this->get_coi_object();
+		$string       = 'hello world';
+		$start_string = 'world';
+		$result       = $this->call_private_method( $object, 'starts_with', [ $string, $start_string ] );
+		$this->assertFalse( $result );
+	}
+
+	/**
+	 * @covers ::is_edit_screen
+	 * @covers ::get_current_screen
+	 */
+	public function test_is_edit_screen() {
+		$object                    = $this->get_coi_object();
+		$GLOBALS['current_screen'] = convert_to_screen( \Google\Web_Stories\Story_Post_Type::POST_TYPE_SLUG );
+		$result                    = $this->call_private_method( $object, 'is_edit_screen' );
+		$this->assertTrue( $result );
+
+		unset( $GLOBALS['current_screen'] );
+	}
+
+	/**
+	 * @covers ::get_current_screen
+	 */
+	public function test_get_current_screen() {
+		$object = $this->get_coi_object();
+		$result = $this->call_private_method( $object, 'get_current_screen' );
+		$this->assertFalse( $result );
+	}
+
+	/**
+	 * @covers ::replace_in_dom
+	 */
+	public function test_replace_in_dom() {
+		$html   = file_get_contents( __DIR__ . '/../../data/cross_origin_content.html' );
+		$html   = str_replace( '--SITE_URL--', site_url(), $html );
+		$object = $this->get_coi_object();
+		$result = $this->call_private_method( $object, 'replace_in_dom', [ $html ] );
+		$this->assertContains( '<script async="" crossorigin="anonymous" src="https://cdn.ampproject.org/v0.js"></script>', $result );
+		$this->assertContains( '<script async="" crossorigin="anonymous" src="https://cdn.ampproject.org/v0/amp-story-1.0.js" custom-element="amp-story"></script>', $result );
+		$this->assertContains( '<link crossorigin="anonymous" href="https://fonts.googleapis.com/css2?display=swap&#038;family=Roboto" rel="stylesheet" />', $result );
+		$this->assertContains( '<img alt="test" crossorigin="anonymous" src="http://www.example.com/test1.jpg" loading="eager" />', $result );
+		$this->assertContains( "<img crossorigin='anonymous' src='http://www.example.com/test2.jpg' alt='test' />", $result );
+		$this->assertContains( '<iframe crossorigin="anonymous" src="http://www.example.com"></iframe>', $result );
+		$this->assertContains( 'crossorigin="use-credentials"', $result );
+		$this->assertContains( '<a href="http://www.example.com/test1.jpg">Test</a>', $result );
```

Comment:
A bit surprised here, after reading the code I thought this would fail. Why doesn't it fail? 🤔

Reply:
I made sure to add the same url as well, to be sure. The reason it doesn't fail here is attribute. The replacement is on `src="http://www.example.com/test1.jpg"`. The worse thing that might happen here is this a tag like iframe with src might replaced. But that isn't an issue, as that needed to be changed anyway.

I can add some more test cases if you are worried.

Reply:
> The reason it doesn't fail here is attribute. The replacement is on `src="http://www.example.com/test1.jpg"`

But according to the map the replacement is also done on `href`? 🤔 

```php
$map = [
	  'link'   => 'href',
	  'img'    => 'src',
	  'iframe' => 'src',
	  'script' => 'src',
];
```

Reply:
That would only happen if this is what the markup looks like 
```
<html>
<head>
<link href="http://www.google.com/test.jpg" ref="test" />
</head>
<body>
<a href="http://www.google.com/test.jpg">Test</a>
</body>
</html>
```

It checks for both attribute and url. The worse that will happen here is you will get a `<a href="http://www.google.com/test.jpg" crossorigin>Test</a>`, which is invalid, but will likely not break anything. 

After changing the avatar, script and style tags, the odds this code doing anything a pretty unlikely. If they are putting image tags and hyperlinks on the page, where would they even render, what would be be breaking here?

Reply:
Thanks for the explanation!

Seems like an acceptable low risk in that case.

Let's see how it works.